openpype/hosts/maya/startup/userSetup.py

```diff
@@ -1,16 +1,67 @@
 import os
+import shutil
+
 from openpype.settings import get_project_settings
 from openpype.pipeline import install_host
-from openpype.hosts.maya.api import MayaHost
+from openpype.hosts.maya.api import MayaHost, current_file
+from openpype.lib import register_event_callback
+
 from maya import cmds
 
 host = MayaHost()
 install_host(host)
 
 
-print("starting OpenPype usersetup")
+print("Starting OpenPype usersetup...")
+
+# Open Workfile Post Initialization.
+key = "OPENPYPE_OPEN_WORKFILE_POST_INITIALIZATION"
+if bool(int(os.environ.get(key, "0"))):
+    cmds.evalDeferred(
+        "cmds.file(os.environ[\"AVALON_LAST_WORKFILE\"], open=True,"
+        " force=True)",
+        lowestPriority=True
+    )
```

Comment:
Eval deferred is capable of running from an actual function too, for example:
```python
from maya import cmds

def hello_world():
    print('hello')
    
cmds.evalDeferred(hello_world)
```

Thus here we could also do:
```python
from functools import partial
cmds.evalDeferred(partial(cmds.file, 
                          os.environ["AVALON_LAST_WORKFILE"],
                          open=True,
                          force=True),
                  lowestPriority=True)
```

Reply:
> Also, I think there's a bug here @tokejepsen
> 
> If you disable launching last workfiles automatically globally but enable `maya/open_workfile_post_initialization` then it'd still try to read the `AVALON_LAST_WORKFILE` which I think might not exist - and even if it does.. it shouldn't load the file automatically because the pipeline had it disabled.

The logic for whether to open the last workfile is handled in [here](https://github.com/tokejepsen/pype/blob/feature/OP-3074_xgen/openpype/hooks/pre_add_last_workfile_arg.py#L36), so the enviroment variable `OPENPYPE_OPEN_WORKFILE_POST_INITIALIZATION` is never set and no workfile is launched.
Should this be more explicit and printed/logged?

Reply:
I see, you're correct :) As long as it works I'm fine with it.